internal/appsec/emitter/sharedsec/actions.go

```diff
@@ -44,17 +48,24 @@ func init() {
 }
 
 type (
-	// Action represents a WAF action.
-	// It holds the HTTP and gRPC handlers to be used instead of the regular
-	// request handler when said action is executed.
-	Action struct {
-		http     http.Handler
-		grpc     GRPCWrapper
-		blocking bool
+	// Action is a generic interface that represents any WAF action
+	Action interface {
+		Blocking() bool
+		EmitData(op dyngo.Operation)
 	}
 
-	// Actions represents a set of action bindings to an action name.
-	Actions map[string]*Action
+	// HTTPAction are actions that interact with an HTTP request flow (block, redirect...)
+	HTTPAction struct {
+		http.Handler
+	}
+	// GRPCAction are actions that interact with a GRPC request flow
+	GRPCAction struct {
+		GRPCWrapper
+	}
```

Comment:
I'd move those to httpsec and grpcsec

Reply:
Will do in another PR if possible. I got started on this and it's getting bigger and bigger